### PR TITLE
Fixes OpenXR

### PR DESCRIPTION
**Pull request by MaximilianEmel**
# PR Details

- Fixes OpenXR crash
- Fixes OpenXR mirroring
- Removes Remnants of the deprecated Fove & GoogleVR APIs

## Description

- Upgrades Silk.NET.OpenXR from 2.15.0 to 2.17.1

## Related Issue

https://github.com/stride3d/stride/issues/1909

## Motivation and Context

It fixes OpenXR, and cleans things up.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My change requires a change to the documentation.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: sources/engine/Stride.VirtualReality/Stride.VirtualReality.csproj
Comment (Eideren) [inline sources/engine/Stride.VirtualReality/Stride.VirtualReality.csproj:55]: We should update all Silk.NET versions referenced in the solution together (see `Stride.Graphics.csproj`) otherwise Perksey's gonna kill us again

File: sources/engine/Stride.Engine/Rendering/Compositing/ForwardRenderer.cs
Comment (Eideren) [inline sources/engine/Stride.Engine/Rendering/Compositing/ForwardRenderer.cs:729]: I understand this is to fix the mirror issue you mentioned, why is this necessary exactly ?

**Review by Eideren — COMMENTED** [review]
Thanks for the PR !

File: sources/engine/Stride.VirtualReality/Stride.VirtualReality.csproj
Comment (tebjan) [reply sources/engine/Stride.VirtualReality/Stride.VirtualReality.csproj:55]: image

File: sources/engine/Stride.Engine/Rendering/Compositing/ForwardRenderer.cs
Comment (tebjan) [reply sources/engine/Stride.Engine/Rendering/Compositing/ForwardRenderer.cs:729]: Also, is this fix influencing anything when running OpenVR?

Comment (MaximilianEmel) [reply sources/engine/Stride.Engine/Rendering/Compositing/ForwardRenderer.cs:729]: It's necessary because OpenXR doesn't have MirrorTexture implemented, and the OpenVR implementation is broken anyway. The whole logic there was flawed to begin with, because it meant that if you enabled PostFx, it would magically enable the Mirror, even if you had disabled it.

## Here's the results of all combinations for reference:
### VRSettings.VRDevice.MirrorTexture:
OpenXR + PostFx: Error on Startup (MirrorTexture doesn't exist)
OpenXR + no PostFx: Error on Startup (MirrorTexture doesn't exist)
OpenVR + PostFX: Bad crop - shows only upper left of each eye on Mirror
OpenVR + no PostFx: Bad crop - shows only upper left of each eye on Mirror

### vrFullSurface:
OpenXR + PostFx: Good
OpenXR + no PostFx: Good
OpenVR + PostFX: Good
OpenVR + no PostFx: Good

File: sources/engine/Stride.VirtualReality/Stride.VirtualReality.csproj
Comment (MaximilianEmel) [reply sources/engine/Stride.VirtualReality/Stride.VirtualReality.csproj:55]: Done.

**Review by Eideren — APPROVED** [review]

**Comment by Eideren** [comment]
LGTM but @tebjan has far more experience with VR stuff, so just waiting on his approval.

**Comment by tebjan** [comment]
@arturoc would you have time to take a quick look at this?

**Comment by arturoc** [comment]
Yes, I've been busy these last few days but can take a look next week

**Comment by Eideren** [comment]
Don't have any XR to test this out but OpenVR still works as expected. @arturoc seems quite busy, so I say we merge this in. It can't be worse than the current state of XR anyway.

**Comment by Eideren** [comment]
Thanks, and sorry for the delay @MaximilianEmel !

**Comment by Basewq** [comment]
FYI, this pull broke the OpenGL code because Silk.NET had breaking changes from 2.15.0 to 2.17.1
Opening the Stride.Android.sln will be the easiest way to see what broke.

**Pull request merged**
